src/devices/cpu/h8/h8_sci.cpp

```diff
@@ -192,54 +192,51 @@ void h8_sci_device::clock_update()
 		else
-			clock_mode = CLKM_INTERNAL_ASYNC;
+			clock_mode = ClockMode::INTERNAL_ASYNC;
 	}
 
-	if(clock_mode == CLKM_EXTERNAL_ASYNC && !external_clock_period.is_never())
-		clock_mode = CLKM_EXTERNAL_RATE_ASYNC;
-	if(clock_mode == CLKM_EXTERNAL_SYNC && !external_clock_period.is_never())
-		clock_mode = CLKM_EXTERNAL_RATE_SYNC;
+	if(clock_mode == ClockMode::EXTERNAL_ASYNC && !external_clock_period.is_never())
+		clock_mode = ClockMode::EXTERNAL_RATE_ASYNC;
+	if(clock_mode == ClockMode::EXTERNAL_SYNC && !external_clock_period.is_never())
+		clock_mode = ClockMode::EXTERNAL_RATE_SYNC;
 
 	if(V>=1) {
-		char buf[4096];
+		std::string new_message;
 		switch(clock_mode) {
-		case CLKM_INTERNAL_ASYNC:
-			sprintf(buf, "clock internal at %d Hz, async, bitrate %d bps\n", int(cpu->clock() / divider), int(cpu->clock() / (divider*16)));
+		case ClockMode::INTERNAL_ASYNC:
+			new_message = util::string_format("clock internal at %d Hz, async, bitrate %d bps\n", int(cpu->clock() / divider), int(cpu->clock() / (divider*16)));
 			break;
-		case CLKM_INTERNAL_ASYNC_OUT:
-			sprintf(buf, "clock internal at %d Hz, async, bitrate %d bps, output\n", int(cpu->clock() / divider), int(cpu->clock() / (divider*16)));
+		case ClockMode::INTERNAL_ASYNC_OUT:
+			new_message = util::string_format("clock internal at %d Hz, async, bitrate %d bps, output\n", int(cpu->clock() / divider), int(cpu->clock() / (divider*16)));
 			break;
-
-		case CLKM_EXTERNAL_ASYNC:
-			sprintf(buf, "clock external, async\n");
+		case ClockMode::EXTERNAL_ASYNC:
+			new_message = "clock external, async\n";
 			break;
-		case CLKM_EXTERNAL_RATE_ASYNC:
-			sprintf(buf, "clock external at %d Hz, async, bitrate %d bps\n", int(cpu->clock()*internal_to_external_ratio), int(cpu->clock()*internal_to_external_ratio/16));
+		case ClockMode::EXTERNAL_RATE_ASYNC:
+			new_message = util::string_format("clock external at %d Hz, async, bitrate %d bps\n", int(cpu->clock()*internal_to_external_ratio), int(cpu->clock()*internal_to_external_ratio/16));
 			break;
-
-		case CLKM_INTERNAL_SYNC_OUT:
-			sprintf(buf, "clock internal at %d Hz, sync, output\n", int(cpu->clock() / (divider*2)));
+		case ClockMode::INTERNAL_SYNC_OUT:
+			new_message = util::string_format("clock internal at %d Hz, sync, output\n", int(cpu->clock() / (divider*2)));
 			break;
-
-		case CLKM_EXTERNAL_SYNC:
-			sprintf(buf, "clock external, sync\n");
+		case ClockMode::EXTERNAL_SYNC:
+			new_message = "clock external, sync\n";
 			break;
-		case CLKM_EXTERNAL_RATE_SYNC:
-			sprintf(buf, "clock external at %d Hz, sync\n", int(cpu->clock()*internal_to_external_ratio));
+		case ClockMode::EXTERNAL_RATE_SYNC:
+			new_message = util::string_format("clock external at %d Hz, sync\n", int(cpu->clock()*internal_to_external_ratio));
 			break;
 		}
-		if(buf != last_clock_message) {
-			last_clock_message = buf;
-			logerror("%s", buf);
+		if(new_message != last_clock_message) {
+			last_clock_message = new_message;
+			logerror(new_message);
```

Comment:
You should `std::move` the new message into `last_clock_message` to avoid an unnecessary copy if the message is too large for small string optimisation.

Reply:
Done